src/platforms/platform.dom.js

```diff
@@ -312,29 +313,34 @@ function injectCSS(rootNode, css) {
 	}
 }
 
-export default {
-	/**
-	 * When `true`, prevents the automatic injection of the stylesheet required to
-	 * correctly detect when the chart is added to the DOM and then resized. This
-	 * switch has been added to allow external stylesheet (`dist/Chart(.min)?.js`)
-	 * to be manually imported to make this library compatible with any CSP.
-	 * See https://github.com/chartjs/Chart.js/issues/5208
-	 */
-	disableCSSInjection: false,
-
+/**
+ * Platform class for charts that can access the DOM and global window/document properties
+ * @extends Platform
+ */
+export default class DomPlatform extends Platform {
 	/**
-	 * This property holds whether this platform is enabled for the current environment.
-	 * Currently used by platform.js to select the proper implementation.
-	 * @private
+	 * @constructor
+	 * @param {object} options - The chart options
 	 */
-	_enabled: typeof window !== 'undefined' && typeof document !== 'undefined',
+	constructor(options) {
+		super();
+
+		/**
+		 * When `true`, prevents the automatic injection of the stylesheet required to
+		 * correctly detect when the chart is added to the DOM and then resized. This
+		 * switch has been added to allow external stylesheet (`dist/Chart(.min)?.js`)
+		 * to be manually imported to make this library compatible with any CSP.
+		 * See https://github.com/chartjs/Chart.js/issues/5208
+		 */
+		this.disableCSSInjection = options.disableCSSInjection;
```

Comment:
I just realized while using this setting on another project that it's a little bit weird to make this a per-chart setting. You either include the stylesheet in the page manually once or you inject it into the page once. It doesn't really make sense to make it different on a per-chart basis. Does this help with the web-worker somehow or would it be possible to change it back? Sorry for the additional comment late in the review cycle

Reply:
Yeah, that makes sense. I can move this config to be global to Chartjs. I think the right way to do it might be do restore `platform` to be a global object, make `disableCSSInjection` a property on that, make `platforms` a property of `platform`, and rename the `Platform` class to `BasePlatform` for clarity.

Reply:
I'd leave `platforms` on `Chart`. The only reason it's needed is for users who use the compiled version of Chart.js. But if you're grabbing from npm and building into the project yourself we wouldn't need it. I'd like to keep all those types of inclusions in a single place so that we can handle them more easily if we add better tree shaking support.

I can go either way on having `platform` be global or not. I'm not sure if there's a usecase for some offscreen and some onscreen canvas

Reply:
I guess I'm not quite sure what you're envisioning. I was thinking something like:
```
Chart.platform.disableCSSInjection = true;
// my proposal
new Chart({ platform: Chart.platform.platforms.BasicPlatform });
// Is this your alternative?
new Chart({ platform: Chart.platforms.BasicPlatform });
```

I think there might be one usecase for having both onscreen and offscreen canvases in the same global Chartjs context: render a chart in an onscreen canvas, have a 'download' button that downloads the same chart in a much larger size or with some edited styling (this could be achieved today by using a hidden canvas element too). I think the reason I like a per-chart platform is that onscreen and offscreen canvases can both exist in the same context and it introduces some awkwardness to make it global - I could see it being confusing.

Reply:
Yes, I was thinking for users that use the `script` include that it'd be the following. I think that's how it would work with this PR now.
```
new Chart({ platform: Chart.platforms.BasicPlatform });
```

A bit of a side note, but for npm users, I'm hoping it would possible to do something like below. Don't worry about it here though.
```
import BasicPlatform from 'chart.js/platforms/platform.basic';
new Chart({ platform: BasicPlatform });
```

I'm totally cool with per-platform chart. This PR's looking good, so I don't think we should change too much. The only thing I was thinking we might consider was where `disableCSSInjection` lives. I guess there's a question of where to put `disableCSSInjection` though since `Chart.platform` doesn't exist anymore. Maybe we could have a `Chart.globalOptions` or something like that, which is declared in `core.controller.js`. What do you think @kurkle @etimberg ?

Reply:
`Chart.defaults.platform`?

Reply:
It's a little weird to call it a default since it's where the user would set the actual value as well. What happens when the user tries to pass `platform.disableCSSInjection` to the chart options? I'm assuming we don't want to allow the user to set it on a per-chart basis